src/ChainSafe.Gaming/Web3/Core/Logout/ILogoutHandler.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace ChainSafe.Gaming.Web3.Core.Logout
+{
+    public interface ILogoutHandler
```

Comment:
"Logout" is not in context for Wallet Connect, this can be `IOnDisconnectedCallbackHandler` or something similar

Reply:
And they can work the same way as `ILifeCycleParticipant` so there's no need for `ILogoutManager`

Reply:
`IOnWeb3TerminatedCallbackHandler` is also not a bad name imo

Reply:
It's not in the context, but it's required for the correct logout handling. WillStopAsync will be called even if a user doesn't log out. When the user logs out, he doesn't get automatically logged in the next time he starts the game. That's the difference between logging out and simply terminating the web3.

Reply:
you can have an overload method or a default variable in `TerminateAsync(bool disconnect = false)` and basically call `IOnDisconnectedCallbackHandler.OnDiconnected` if it's set to true for all registered `IOnDisconnectedCallbackHandler` services and on Unity's side if it's a logout you can call `TerminateAsync(true)` or have a different method called `Disconnect()` in Web3 that calls both `TerminateAsync` and `IOnDisconnectedCallbackHandler`

Reply:
Refactored, but kept the "Logout" name as it is more understandable IMO

Reply:
I mean it's out of context becuase Login isn't what we're actually doing on WC or Metamask, it's connecting to a wallet but I'm okay making compromises on this one